keras/layers/core.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, mask_value=0., **kwargs):
         self.mask_value = mask_value
 
     def compute_mask(self, inputs, mask=None):
-        return K.any(K.not_equal(inputs, self.mask_value), axis=-1)
+        return K.cast(K.any(K.not_equal(inputs, self.mask_value), axis=-1), 'bool')
```

Comment:
Though, the issue with standardizing to `bool` instead of `uint8` is that you have to cast before a multiplication.  You can't do `K.any(x) * K.any(y)`.  Should I switch things to `uint8` instead?